pkg/reconciler/configuration/configuration.go

```diff
@@ -196,64 +212,118 @@ func (c *Reconciler) reconcile(ctx context.Context, config *v1alpha1.Configurati
+func CheckNameAvailability(config *v1alpha1.Configuration, lister listers.RevisionLister) error {
 	// If config.Spec.GetTemplate().Name is set, then we can directly look up
 	// the revision by name.
 	name := config.Spec.GetTemplate().Name
 	if name == "" {
-		return nil, nil
+		return nil
 	}
 	errConflict := errors.NewAlreadyExists(v1alpha1.Resource("revisions"), name)
 
 	rev, err := lister.Revisions(config.Namespace).Get(name)
 	if errors.IsNotFound(err) {
 		// Does not exist, we must be good!
 		// note: for the name to change the generation must change.
-		return nil, err
+		return err
 	} else if err != nil {
-		return nil, err
+		return err
 	} else if !metav1.IsControlledBy(rev, config) {
 		// If the revision isn't controller by this configuration, then
 		// do not use it.
-		return nil, errConflict
+		return errConflict
 	}
 
 	// Check the generation on this revision.
 	generationKey := serving.ConfigurationGenerationLabelKey
 	expectedValue := resources.RevisionLabelValueForKey(generationKey, config)
 	if rev.Labels != nil && rev.Labels[generationKey] == expectedValue {
-		return rev, nil
+		return nil
 	}
 	// We only require spec equality because the rest is immutable and the user may have
 	// annotated or labeled the Revision (beyond what the Configuration might have).
 	if !equality.Semantic.DeepEqual(config.Spec.GetTemplate().Spec, rev.Spec) {
-		return nil, errConflict
+		return errConflict
 	}
-	return rev, nil
+	return nil
 }
 
-func (c *Reconciler) latestCreatedRevision(config *v1alpha1.Configuration) (*v1alpha1.Revision, error) {
-	if rev, err := CheckNameAvailability(config, c.revisionLister); rev != nil || err != nil {
-		return rev, err
+func (c *Reconciler) getSortedCreatedRevisions(ctx context.Context, config *v1alpha1.Configuration) ([]*v1alpha1.Revision, error) {
+	logger := logging.FromContext(ctx)
+	if err := CheckNameAvailability(config, c.revisionLister); err != nil {
+		return nil, err
 	}
 
 	lister := c.revisionLister.Revisions(config.Namespace)
-	generationKey := serving.ConfigurationGenerationLabelKey
 
-	list, err := lister.List(labels.SelectorFromSet(map[string]string{
-		generationKey:                 resources.RevisionLabelValueForKey(generationKey, config),
+	configSelector := labels.SelectorFromSet(map[string]string{
 		serving.ConfigurationLabelKey: config.Name,
-	}))
+	})
 
-	if err == nil && len(list) > 0 {
-		return list[0], nil
+	if config.Status.LatestReadyRevisionName != "" {
+		// Get the latest ready revision
+		latestReadyRev, err := lister.Get(config.Status.LatestReadyRevisionName)
+		if err != nil {
+			return nil, err
+		}
+
+		generationKey := serving.ConfigurationGenerationLabelKey
+		// Generate the slice that contains all the generations we'd like to apply for the "In" filter
+		startInt := latestReadyRev.Generation
+		endInt, err := strconv.Atoi(resources.RevisionLabelValueForKey(generationKey, config))
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Can we expose a method on Revision to get this, if we haven't already got one?

Reply:
I just updated the PR to use config.Generation here directly which is more straightforward